### PR TITLE
Feature: `--env`, `--env-path` flags.

### LINKED ISSUE #131: TOC/Headings - some are incorrect position / level
See `Custom Flags` for example.

### PR DESCRIPTION
**Pull request by ryan-roemer**
* Adds `--env`, `--env-path` flags so you can add cross-OS-compatible environment variables via a JSON object.
* Updates travis ci matrix
* Abstract json parsing to utility.
* Fix indentation level of some sections. Fixes #131 

/cc @exogen

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -20,8 +21,11 @@ test_script:
   # Output useful info for debugging.
   - node --version
   - node_modules\.bin\npm --version
-  # run tests
-  - node_modules\.bin\npm run builder:check
+  # TODO REENABLE
```
Comment (exogen) [inline appveyor.yml]: Add comment on when/why or link to ticket?

File: lib/environment.js [hunk 2]
```diff
@@ -125,3 +134,34 @@ Environment.prototype.updateConfigVars = function (pkgConfigs) {
...
+
+  // Parse envs.
+  var envsObj = jsonParse({
+    str: opts.env,
+    path: opts.envPath
+  });
+
+  // Validation
+  if (_.isEmpty(envsObj)) {
+    throw new Error("Empty/null JSON environment.");
```
Comment (exogen) [inline lib/environment.js]: I could see this being perfectly intentional/allowable behavior, especially if people are running `builder` as part of a larger setup where their scripts just always pass `--env` / `--env-path` but they sometimes contain no overrides.

Comment (ryan-roemer) [inline lib/environment.js]: Good point. I'll remove.

File: appveyor.yml (hunk 1 above)
Comment (ryan-roemer) [inline appveyor.yml]: Haha, it's there precisely to make sure I **don't** leave this in the final PR (I had a lot of windows debugging to do earlier...)

**Comment by ryan-roemer** [comment]
@exogen -- Ready for re-review!

**Pull request merged**